### DISCUSSION TITLE
Support for portable recursive file/folder removal

### DISCUSSION THREAD

**Discussion by mcandre** [libuv/libuv#3529] [Ideas]
Hi,

I really like how libuv makes most filesystem operations easy and portable (similar to the C++ `filesystem` library)!

One gap I noticed in libuv, however, is an apparent lack of support for ensuring that a file or folder is deleted (the `std::filesystem::remove_all` function).

Can we get a portable remove_all equivalent for C users?

POSIX can do most of the heavy lifting there:

https://stackoverflow.com/a/5467788

Though need to test on directory and non-directory file paths.

And need to find out the (COMSPEC) Windows equivalent calls.

**Comment by vtjnash** [comment]
I think libuv is not that kind of library. It is not intended to be a replacement for the C stdlib. It is a kernel-interface layer for programs. though someone could therefore use it as a platform to build a bigger compiler support library.
